src/Illuminate/Database/Connection.php

```diff
@@ -1612,6 +1614,56 @@ public function getServerVersion(): string
         return $this->getPdo()->getAttribute(PDO::ATTR_SERVER_VERSION);
     }
 
+    /**
+     * Determine if the connection matches the given driver name and version.
+     *
+     * @template TValue
+     *
+     * @param  string|array|\Closure(string,string): TValue  $driver
+     * @param  string|null  $operator
+     * @param  string|null  $version
```

Comment:
Would having explicit methods help DX better? 

```php
$c->isMariaDB();
$c->isMySql();
...
```

Reply:
Yes, I know about macros. But, it would be nicer if the framework already enforce what DB it has support to since a `string` can be anything. ✌🏼

Reply:
Currently, we need this dynamicity on the core.